api/v1beta1/disruption_webhook.go

```diff
@@ -99,6 +104,22 @@ func (r *Disruption) ValidateCreate() error {
 		return errors.New("the chaos handler is disabled but the disruption onInit field is set to true, please enable the handler by specifying the --handler-enabled flag to the controller if you want to use the onInit feature (requires Kubernetes >= 1.15)")
 	}
 
+	if r.Spec.Network.Cloud != nil {
+		clouds := map[cloudtypes.CloudProviderName][]string{
+			cloudtypes.CloudProviderAWS:     r.Spec.Network.Cloud.AWS,
+			cloudtypes.CloudProviderDatadog: r.Spec.Network.Cloud.Datadog,
+			cloudtypes.CloudProviderGCP:     r.Spec.Network.Cloud.GCP,
+		}
+
+		for cloudName, serviceList := range clouds {
+			for _, service := range serviceList {
+				if !cloudProviderManager.ServiceExists(cloudName, service) {
+					return fmt.Errorf("service %s of %s does not exist. Available are: %s", service, cloudName, strings.Join(cloudProviderManager.GetServiceList(cloudName), ", "))
+				}
+			}
+		}
+	}
+
```

Comment:
Added this in disruption_webhook.go instead of the specific Validate() method of the NetworkDisruptionSpec [here](https://github.com/DataDog/chaos-controller/blob/d934f1534a70126846d2f81493147e3777bce6c8/api/v1beta1/network_disruption.go#L104) due to a problem with the package deepcopy-gen. It's preventing us from passing the cloudProviderManager to those methods because we can't generate the DeepCopyInto method on interfaces.
In my opinion, we should put this in Validate() but I'm not sure the time wasted trying to find a workaround is worh it.